src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1405,6 +1406,216 @@ impl ToQemuParams for QmpSocket {
     }
 }
 
+// DeviceDriver is the device driver string.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(dead_code)]
+pub enum DeviceDriver {
+    PCIBridge,
+    PCIePCIBridge,
+    RootPort,
+    SwitchUpstreamPort,
+    SwitchDownstreamPort,
+}
+
+impl std::fmt::Display for DeviceDriver {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let driver = match self {
+            DeviceDriver::PCIBridge => "pci-bridge",
+            DeviceDriver::PCIePCIBridge => "pcie-pci-bridge",
+            DeviceDriver::RootPort => "pcie-root-port",
+            DeviceDriver::SwitchUpstreamPort => "x3130-upstream",
+            DeviceDriver::SwitchDownstreamPort => "xio3130-downstream",
+        };
+        write!(f, "{}", driver)
+    }
+}
+
+// PCIeRootPortDevice directly attached onto the root bus
+#[derive(Debug, Default)]
+pub struct PCIeRootPortDevice {
+    pub id: String,
+    pub bus: String,
+    pub chassis: String,
+    pub slot: String,
+    pub multifunction: bool,
+    pub addr: String,
+    pub bus_reserve: String,
+    pub pref64_reserve: String,
+    pub pref32_reserve: String,
+    pub mem_reserve: String,
+    pub io_reserve: String,
+}
+
+#[async_trait]
+impl ToQemuParams for PCIeRootPortDevice {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut qemu_params = Vec::new();
+        let mut device_params = Vec::new();
+
+        device_params.push(format!("{},id={}", DeviceDriver::RootPort, self.id));
+
+        let bus = if self.bus.is_empty() {
+            "pcie.0"
```

Comment:
Please make this a constant. Also how does it relate with other occurrences of `"pcie.0"` in https://github.com/Apokleos/kata-containers/blob/vfio-hotplug-runtime-rs/src/runtime-rs/crates/hypervisor/src/device/topology.rs ?